src/Lykke.Logs/SpamGuard.cs

```diff
@@ -41,32 +41,43 @@ internal void SetMutePeriod(LogLevel level, TimeSpan mutePeriod)
             if (!_mutePeriods.ContainsKey(level))
                 return false;
 
+            var levelDict = _lastMessages.GetOrAdd(level, new Dictionary<string, DateTime>());
+            var key = GetEntryKey(component, process);
+            var now = DateTime.UtcNow;
+            DateTime lastTime = now;
+            lock(levelDict)
```

Comment:
This and another lock of leveDict are both very short (no IO, only CPU inside lock), in this case SemaphorSlim could give significant performance gain. Consider this option, please